pkg/store/store.go

```diff
@@ -424,36 +425,23 @@ func (s *Store) LoadTxsInBlock(ctx context.Context, blockHeight int64) ([]models
 	return txs, nil
 }
 
-// LoadTxsBySourceAndDest
-func (s *Store) LoadTxsBySourceAndDest(ctx context.Context, source, dest string) ([]models.Transaction, error) {
-	var rows *sql.Rows
-	var err error
-	switch {
-	case source == "" && dest == "":
-		return nil, errors.Wrap(errors.ErrInput, "source and destination empty")
-	case source == "" && dest != "":
-		rows, err = s.db.QueryContext(ctx, `
-			SELECT transaction_hash, block_id, message
-			FROM transactions
-			WHERE message ->> 'path' = 'cash/send' 
-			AND message -> 'details' ->> 'destination' = $1
-		`, dest)
-	case source != "" && dest == "":
-		rows, err = s.db.QueryContext(ctx, `
-			SELECT transaction_hash, block_id, message
-			FROM transactions
-			WHERE message ->> 'path' = 'cash/send' 
-			AND message -> 'details' ->> 'source' = $1
-		`, source)
-	case source != "" && dest != "":
-		rows, err = s.db.QueryContext(ctx, `
-			SELECT transaction_hash, block_id, message
-			FROM transactions
-			WHERE message ->> 'path' = 'cash/send' 
-			AND message -> 'details' ->> 'source' = $1
-			AND message -> 'details' ->> 'destination' = $2
-		`, source, dest)
+func (s *Store) LoadTxsByParams(ctx context.Context, source, dest, memo string) ([]models.Transaction, error) {
+	if source == "" && dest == "" && memo == "" {
+		return nil, errors.Wrap(errors.ErrInput, "no input provided")
 	}
+
+	query := "SELECT transaction_hash, block_id, message FROM transactions"
```

Comment:
I need to build a dynamic(chained) query that uses 3 optional values. How to achieve this using "database/sql" @husio ?

Reply:
I don't understand what you want to achieve. Can you explain more?

Reply:
SQL injection attacks came to my mind. I was trying stuff around

Reply:
What I wrote is incorrect. I will update the query soon.